### PR TITLE
drainer: load infos from dump files

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: close #xxx

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `tidb-ansible` repository
 - Need to be included in the release note

### Release note

- No release note

### CODE REVIEW

File: drainer/util.go [hunk 1]
```diff
@@ -191,3 +201,183 @@ func genDrainerID(listenAddr string) (string, error) {
...
+
+// getTableFromDumpFilename extracts db and table name from dump filename.
+func getTableFromDumpFilename(filename string) (db, table string, ok bool) {
+	if !strings.HasSuffix(filename, "-schema.sql") {
+		return "", "", false
+	}
+
+	idx := strings.LastIndex(filename, "-schema.sql")
+	name := filename[:idx]
+	fields := strings.Split(name, ".")
```
Comment (lichunzhu) [inline drainer/util.go:259]: https://github.com/pingcap/dumpling/issues/131
Maybe we need to make sure there table name not contain special characters, or we have to parse it.

File: drainer/schema.go [hunk 2]
```diff
@@ -484,3 +492,174 @@ func addImplicitColumn(table *model.TableInfo) {
...
+		}
+		if job, err := s.mockCreateTableJob(info.stmt, dbInfo.id, info.id, v); err != nil {
+			log.Error("fail to handle create table", zap.Error(err))
+			return err
+		} else if _, _, _, err = s.handleDDL(job); err != nil {
+			return err
+		}
+		v++
+	}
+	return nil
```
Comment (lichunzhu) [inline drainer/schema.go:546]: Add this to make sure these memory can be garbage collected?
```suggestion
	// safe for gc
	s.dbInfos = nil
	s.tbInfos = nil
	return nil
```

**Review by GMHDBJD — COMMENTED** [review]
LGTM

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**